eng/mgmt/automation/utils.py

```diff
@@ -350,3 +353,83 @@ def set_or_increase_version(
 
 def is_windows():
     return platform.system().lower() == "windows"
+
+
+def get_latest_release_version(previous_version: str, current_version: str) -> str:
+    if "-beta." in current_version and "-beta." not in previous_version:
+        # if current version is preview, try compare it with a previous preview release
+
+        version_pattern = r"\d+\.\d+\.\d+-beta\.(\d+)?"
+        beta_version_int = int(re.match(version_pattern, current_version).group(1))
+        if beta_version_int > 1:
+            previous_beta_version_int = beta_version_int - 1
+            previous_beta_version = current_version.replace(
+                "-beta." + str(beta_version_int),
+                "-beta." + str(previous_beta_version_int),
+            )
+            previous_version = previous_beta_version
+    return previous_version
+
+
+def get_latest_ga_version(group_id: str, module: str, previous_version: str) -> str:
+    """
+    If previous_version is GA, return previous_version,
+    Otherwise, return previous GA version before previous beta.
+    previous_version itself if no previous GA version.
```

Comment:
Why return non-GA version, if the method is get_latest_ga_version?

Reply:
Good point. This was for later check in `compare_with_maven_package`. 
Though a second thought, this should only return GA version, and None if no GA version found. Later check should check for None also.

Reply:
Fixed and tested locally.